src/Pipeline.ts

```diff
@@ -0,0 +1,108 @@
+import Commander from 'ioredis/built/utils/Commander';
+import {Command} from 'ioredis';
+import {MomentoRedisAdapter} from './momento-redis-adapter';
+import {ArgumentType} from 'ioredis/built/Command';
+import {CommandParameter} from 'ioredis/built/types';
+
+interface Pipeline {
+  length: number;
+}
+
+class Pipeline extends Commander<{type: 'pipeline'}> {
```

Comment:
Let's document the architectural purpose of this class, how it is used internally and the flow of the important methods we're overriding. For $otherDev and $futureSelf 😄